File: 프로그래머스/3/42579. 베스트앨범/베스트앨범.py
def solution(genres, plays):
    answer = []
    
    dic = {}

    for i in range(len(genres)):
        if genres[i] in dic:
            dic[genres[i]].append([plays[i],i])
        else:
            dic[genres[i]] = [[plays[i],i]]
    
    
    arr = []
    for i in dic:
        arr.append(dic[i])
    

    for i in range(len(arr)):
        arr[i].sort(key=lambda x: (-x[0],x[1]))


    
    count = [0]*len(arr)
    for i in range(len(arr)):
        for j in range(len(arr[i])):
            count[i] += arr[i][j][0]


    
    
    for i in range(len(count)):
        m = count.index(max(count))
        
        for j in range(2):
            try:
                answer.append(arr[m][j][1])
            except IndexError:
                break
        count.pop(m)
        arr.pop(m)
    
    return answer

File: 프로그래머스/3/42579. 베스트앨범/베스트앨범.swift
import Foundation

func solution(_ genres:[String], _ plays: [Int]) -> [Int] {
    var dic = [String: [[Int]]]()
    var cnt = [String: Int]()
    
    for i in 0..<genres.count {
        if let genre = dic[genres[i]] {
            dic[genres[i]]?.append([i,plays[i]])
            cnt[genres[i]]? += plays[i]
        } else {
            dic[genres[i]] = [[i,plays[i]]]
            cnt[genres[i]] = plays[i]
        }
    }
    
    for key in dic.keys {
        dic[key]?.sort(by: { $0[1] > $1[1] })
    }
    let ord = Array(cnt.sorted(by: { $0.value > $1.value }))
    print(ord)
    
    var res = [Int]()
    for genre in ord {
        print(genre)
        for i in 0..<2 {
            if i >= dic[genre.key]?.count ?? 0 { break }
            res.append(dic[genre.key]?[i][0] ?? -1)
            
        }
    }
    return res
}